RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowClusterizer.cc

```diff
@@ -181,6 +200,15 @@ growPFClusters(const reco::PFCluster& topo,
 	  << "Warning! :: pfcluster-topocell distance is too large! d= "
 	  << d2;
       }
+
+      double recHitEnergyNorm=0.;
+
+      for (unsigned int j=0; j<recHitEnergyNormV.size(); ++j) {
+	if((cell_layer == PFLayer::HCAL_BARREL1 || cell_layer == PFLayer::HCAL_ENDCAP) && refhit->depth()!=recHitDepthV[j]) continue;
+	recHitEnergyNorm = recHitEnergyNormV[j];
+      }
```

Comment:
all information computed in this for loop over recHitEnergyNormV is known from values outside of the outer ``` for( auto& cluster : clusters ) ``` loop. Please move the computation outside